### PR TITLE
Update configurations.xml to fix build errors for MPLABX 5.40

### PR DESCRIPTION
**Pull request by yourslab**
Update configurations.xml to fix build errors for MPLABX 5.40

Description
-----------

In the latest release of amazon-freertos (2020007.0):
MPLABX 5.40 DOES NOT work with XC compiler version 2.41, 2.40, and 2.30 on both my Windows and Mac environment.
However, MPLABX 5.35 works with XC compiler version 2.41, 2.40, and 2.30 on both my Windows and Mac environment.
I believe CI is using MPLABX 5.35, and both XC compiler versions 2.40 and 2.30.

The error in MPLABX 5.40 of amazon-freertos 2020007.0 stems from passing a string literal as a preprocessor macro in `configurations.xml`:
```xml
<property key="preprocessor-macros" value="MBEDTLS_CONFIG_FILE=\\\&quot;aws_mbedtls_config.h\\\&quot;;
```
This ends up giving the following error in MPLABX 5.40:
```
In file included from ../../../../../libraries/c_sdk/standard/serializer/src/json/iot_serializer_json_encoder.c:40:0:
../../../../../libraries/3rdparty/mbedtls/include/mbedtls/base64.h:30:10: error: #include expects "FILENAME" or <FILENAME>
#include MBEDTLS_CONFIG_FILE
```

The fix is to update `configurations.xml` to use a single backslash instead:
```xml
<property key="preprocessor-macros" value="MBEDTLS_CONFIG_FILE=\&quot;aws_mbedtls_config.h\&quot;;
```

Keep in mind that I do not experience the error above in version 202002.00 of amazon-freertos as there is no preprocessor macro for `MBEDTLS_CONFIG_FILE`.

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/microchip/curiosity_pic32mzef/mplab/bootloader/nbproject/configurations.xml [hunk 1]
```diff
@@ -190,6 +190,8 @@
       <packs>
         <pack name="PIC32MZ-EF_DFP" vendor="Microchip" version="1.0.20"/>
```
Comment (pavanmr94) [inline projects/microchip/curiosity_pic32mzef/mplab/bootloader/nbproject/configurations.xml:191]: can we change all mchp projects to to use DFP version 1.2.51? that's the latest installed by default by mplab.

Comment (yourslab) [reply projects/microchip/curiosity_pic32mzef/mplab/bootloader/nbproject/configurations.xml:191]: sure!

**Pull request merged**